src/Internal/Address.purs

```diff
@@ -41,26 +37,25 @@ addressToOgmiosAddress :: Address -> Ogmios.OgmiosAddress
 addressToOgmiosAddress = addressBech32
 
 --------------------------------------------------------------------------------
--- `Address` to `ScriptHash`
+-- `Address` to `ValidatorHash`
 --------------------------------------------------------------------------------
--- | Get the `ScriptHash` with an internal `Address`
+
 enterpriseAddressScriptHash :: Address -> Maybe ScriptHash
 enterpriseAddressScriptHash =
   stakeCredentialToScriptHash
     <=< pure <<< enterpriseAddressPaymentCred
     <=< enterpriseAddressFromAddress
 
--- | Get the `ValidatorHash` with an internal `Address`
-enterpriseAddressValidatorHash :: Address -> Maybe ValidatorHash
-enterpriseAddressValidatorHash =
-  map ValidatorHash <<< enterpriseAddressScriptHash
+baseAddressScriptHash :: Address -> Maybe ScriptHash
+baseAddressScriptHash =
+  stakeCredentialToScriptHash
+    <=< pure <<< baseAddressDelegationCred
+    <=< baseAddressFromAddress
 
--- | Get the `MintingPolicyHash` with an internal `Address`
-enterpriseAddressMintingPolicyHash :: Address -> Maybe MintingPolicyHash
-enterpriseAddressMintingPolicyHash =
-  map MintingPolicyHash <<< enterpriseAddressScriptHash
+addressScriptHash :: Address -> Maybe ScriptHash
+addressScriptHash addr =
+  baseAddressScriptHash addr <|> enterpriseAddressScriptHash addr
 
--- | Get the `StakeValidatorHash` with an internal `Address`
-enterpriseAddressStakeValidatorHash :: Address -> Maybe StakeValidatorHash
-enterpriseAddressStakeValidatorHash =
-  map StakeValidatorHash <<< enterpriseAddressScriptHash
+-- | Get the `ValidatorHash` of an address (base or enterprise).
+addressValidatorHash :: Address -> Maybe ValidatorHash
+addressValidatorHash = map ValidatorHash <<< addressScriptHash
```

Comment:
There functions don't make much sense, because they use payment component scripts.

Come to think of it, the current behavior could also be very confusing. I definitely want to give it a second thought.

Reply:
There was a bug